package messages.input;

message MotionCapture {
    
    message Point {
        required string identifier = 1;
        required double size = 2;
        required double x = 3;
        required double y = 4;
        required double z = 5;
    };

    message RigidBody {
        required string identifier = 1;
        required double mean_error = 2;

        // The matrix
        required double xx = 3;
        required double xy = 4;
        required double xz = 5;
        required double xt = 6;
        required double yx = 7;
        required double yy = 8;
        required double yz = 9;
        required double yt = 10;
        required double zx = 11;
        required double zy = 12;
        required double zz = 13;
        required double zt = 14;
        required double tx = 15;
        required double ty = 16;
        required double tz = 17;
        required double tt = 18;

        repeated Point points = 19;
    };

    message Bone {
        optional uint32 parent = 1;
        required RigidBody object = 2;
    };

    message Skeleton {
        required string identifier = 1;
        repeated Bone bones = 3;
    };

    repeated Point points = 1;
    repeated RigidBody rigid_bodies = 2;
    repeated Skeleton skeletons = 3;
}